### PR TITLE
Notebooks selector app with manual GitHub Pages deploy

### PR DESCRIPTION
**Pull request by yatarkan**
Ticket: CVS-119857

This PR introduces OpenVINO Notebooks selector app written with `TypeScript`, `React` and `vite`. Moreover, new GitHub Actions workflow is added for manual app deploy to repository GitHub Pages (through workflow dispatch via GitHub Actions UI).

Note for reviewers: Files in `./selector/src/@spark-design` directory can be skipped during review - these files are from Spark Design open-source package and can be treated as a dependency.

### CODE REVIEW

**Review by aleksandr-mokrov — APPROVED** [review]

File: selector/src/notebook-metadata/notebook-metadata-validator.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 // @ts-check
```
Comment (as-suvorov) [inline selector/src/notebook-metadata/notebook-metadata-validator.js:2]: Consider using json schema instead of manual validation

**Review by as-suvorov — APPROVED** [review]

**Comment by as-suvorov** [comment]
image
I think it would be nice to show tags, tasks on a notebook card

File: selector/src/components/ContentSection/ContentSection.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  const [filteredNotebooksCount, setFilteredNotebooksCount] = useState<number>(0);
+  const [totalNotebooksCount, setTotalNotebooksCount] = useState<number>(0);
+
+  const [itemsPerPage, setItemsPerPage] = useState<number>(notebooksPerPageOptions[0]);
+
+  const totalPages = Math.ceil(filteredNotebooksCount / itemsPerPage);
+
+  useEffect(() => {
+    setPage(1);
+  }, [selectedTags, searchValue, sort, setPage]);
```
[the remark this answers is not in the document]
Comment (yatarkan) [reply selector/src/components/ContentSection/ContentSection.tsx:28]: `setPage` function is used inside this effect so it is considered as a dependency as well

[the remark this answers is not in the document]
Comment (as-suvorov) [reply selector/src/components/ContentSection/ContentSection.tsx:28]: Should `notebooksService.getNotebooks` from the next effect be in dependencies as well?

**Pull request merged**